src/solvers.cxx

```diff
@@ -13,24 +13,24 @@ namespace neml {
 
 // This function is configured by the build
 int solve(Solvable * system, double * x, TrialState * ts,
-          double tol, int miter, bool verbose, bool relative,
-          double * R, double * J)
+          SolverParameters p, double * R, double * J)
 {
 #ifdef SOLVER_NOX
-  return nox(system, x, ts, tol, miter, verbose, R, J);
+  return nox(system, x, ts, p.atol, p.miter, p.verbose, R, J);
 #elif SOLVER_NEWTON
   // Actually selected the newton solver
-  return newton(system, x, ts, tol, miter, verbose, relative, R, J);
+  return newton(system, x, ts, p, R, J);
 #else
   // Default solver: plain NR
-  return newton(system, x, ts, tol, miter, verbose, relative, R, J);
+  return newton(system, x, ts, p, R, J);
 #endif
 }
 
-int newton(Solvable * system, double * x, TrialState * ts,
-          double tol, int miter, bool verbose, bool relative,
-          double * R, double * J)
+int newton(Solvable * system, double * x, TrialState * ts, SolverParameters p, double * R,
+           double * J)
 {
+  int mline = 10;
```

Comment:
didn't notice this before, but shouldn't this bee a controllable parameter?

Reply:
No, not right now.  We're doing really naive backtracking so I don't want them trying to mess with it.  When/if we add proper linesearch we'll want to make it configurable (along with the alpha value or whatever they usually use).

Reply:
fair enough,
you might want to add a TODO comment then, but up to you.

If test passes I'll approve now

Reply:
Thanks.